src/ui/component/stepper/stepperProgress/stepperProgress.tsx

```diff
@@ -0,0 +1,51 @@
+import type { FC } from 'react'
+import { ProgressBar } from './progressBar/progressBar'
+import { getProgressBarColor, getProgressBarTransition } from '../utils'
+import { usePrevious } from '@/ui/hook/usePrevious'
+import './stepperProgress.scss'
+
+type StepId = string
+
+export type StepStatus = {
+  id: StepId
+  status: {
+    valid: boolean
+    active: boolean
+  }
+}
+
+type StepperProgressProps = {
+  stepsStatus: StepStatus[]
+  activeStepIndex: number
+  totalSteps: number
```

Comment:
I think you can tell from the length of the `stepsStatus`, or is it a case where you need it ?